ci-operator/step-registry/idp/freeipa/idp-freeipa-commands.sh

```diff
@@ -0,0 +1,403 @@
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: freeipa
+              key: IPA_DM_PASSWORD
+        - name: SYSTEMD_OFFLINE
+          value: "1"
+        - name: SYSTEMD_NSPAWN_API_VFS_WRITABLE
+          value: "network"
+        volumeMounts:
+          - name: data
+            mountPath: /data
+          - name: systemd-tmp
+            mountPath: /tmp
+          - name: systemd-var-run
+            mountPath: /var/run
+          - name: systemd-var-dirsrv
+            mountPath: /var/run/dirsrv
+      - name: setup-nssdb
+        image: quay.io/freeipa/freeipa-server:fedora-39
+        securityContext:
+          runAsUser: 0
+        command: ["/bin/bash", "-c"]
+        args:
+          - |
+            set -e
+            set -x
+            DATA="/data"
+            NSSDB=/tmp/nssdb-$RANDOM
+            PASSWORD_FILE=/tmp/nssdb-$RANDOM
+            echo $RANDOM > $PASSWORD_FILE
+            openssl pkcs12 -export -in /certs/tls.crt -inkey /certs/tls.key -out "$DATA"/ca.p12 -name "/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA" -passout file:$PASSWORD_FILE
+            pki -d $NSSDB -C $PASSWORD_FILE client-init
+            pki -d $NSSDB pkcs12-cert-mod --pkcs12-file "$DATA"/ca.p12 "/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA" --pkcs12-password-file $PASSWORD_FILE --trust-flags "CTu,Cu,Cu"
+            pk12util -i "$DATA"/ca.p12 -d sql:$NSSDB -w $PASSWORD_FILE -k $PASSWORD_FILE
+            echo -e "$RANDOM\nn\n" | certutil -C -m 2346 -i "$DATA"/ipa.csr -o "$DATA"/ipa.crt -c '/CN=freeipa.freeipa.freeipa.svc.cluster.local/O=FREEIPA' -d $NSSDB -a --extSKID -f $PASSWORD_FILE
+        volumeMounts:
+          - name: data
+            mountPath: /data
+          - name: certs
+            mountPath: /certs
+      containers:
+      - name: freeipa-openshift-container-2-step
+        image: quay.io/freeipa/freeipa-server:fedora-39
+        securityContext:
+          runAsUser: 0
+        tty: true
+        env:
+        - name: DEBUG_TRACE
+          value: ""
+        - name: IPA_SERVER_INSTALL_OPTS
+          value: "--unattended --external-cert-file=/data/ipa.crt --external-cert-file=/certs/tls.crt --realm=IPA.EXAMPLE.COM  --domain=ipa.example.com --no-host-dns"
+        - name: KUBERNETES
+          value: "1"
+        - name: IPA_ADMIN_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: freeipa
+              key: IPA_ADMIN_PASSWORD
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: freeipa
+              key: IPA_DM_PASSWORD
+        - name: SYSTEMD_OFFLINE
+          value: "1"
+        - name: SYSTEMD_NSPAWN_API_VFS_WRITABLE
+          value: "network"
+        ports:
+          - name: http-tcp
+            protocol: TCP
+            containerPort: 80
+          - name: https-tcp
+            protocol: TCP
+            containerPort: 443
+        volumeMounts:
```

Comment:
Because we do not use the http services in our test case, it is no harm to define them, but it is better to define explicitly the ldap 389/636 ports as test case OCP-23334 .